src/com/bishtonsoftware/flaxe/Flaxe.java

```diff
@@ -21,125 +21,112 @@
-
-		// The idea is that then it's easier to modify the actual module being converted
-		// by replacing "PlaceNavigation" without having to keep updating 2 paths which are unlikely to change.
-
-		File baseSrcFolder = new File(args[0]) ;
-		File baseDestFolder = new File(args[1]) ;
-		String folder = args[2] ;
-
-		String act = args[3] ;
-		Action action = null ;
-
-		if (act.equalsIgnoreCase("copy"))
-			action = Action.kCopy ;
-		if (act.equalsIgnoreCase("convert"))
-			action = Action.kConvert;
-
-		if (action == null) {
-			System.err.println("Last parameter must be one of: 'copy', 'convert'; not " + act) ;
-			System.exit(1) ;
-		}
-
-		File srcFolder = new File(baseSrcFolder, folder) ;
-		File destFolder = new File(baseDestFolder, folder) ;
-
-		if (destFolder.exists()) {
-			System.err.println("Refusing to overwrite destination folder '" + destFolder + "'.\n" +
-					           "Please remove it or provide another base destination folder.") ;
-			System.exit(2) ;
-		}
-
-		Flaxe app = new Flaxe() ;
-		app.preProcess(srcFolder, destFolder, action) ;
-		System.out.println("Finished.");
-	}
-
-	private String changeExtension(String original) {
-		return original.replace(".as", ".hx") ;
-	}
-
-	private void preProcess(File srcFolder, File destFolder, Action action) throws IOException {
-		FileFilter filter = FileHelper.createFileFilter(".as") ;
-		List<File> srcFiles = FileHelper.generateFileList(filter, srcFolder, true) ;
-
-		System.out.println("Deleting " + destFolder.getCanonicalPath()) ;
-		destFolder.mkdirs() ;
-		FileHelper.recursiveDelete(destFolder, filter, 0, Long.MAX_VALUE) ;
-
-		for (File src : srcFiles) {
-			String relativePath = FileHelper.getRelativePath(srcFolder, src) ;
-			String modifiedExt = changeExtension(relativePath) ;
-			File dest = new File(destFolder, modifiedExt) ;
-
-			System.out.println(action + " "  + src + " to " + dest) ;
-			convert(src, dest, action) ;
-		}
-	}
-
-	public void convert(File inputFile, File outputFile, Action action) throws IOException {
-		List<String> lines = new ArrayList<>() ;
-		String line ;
-
-		// We read in the entire input file
-		// so that we can examine any part, not just limited to line by line analysis
-		// Using try-with-resource to catch and close
-		try (BufferedReader input = new BufferedReader(new FileReader(inputFile))) {
-			while ((line = input.readLine()) != null) {
-				lines.add(line);
-			}
-		}
-
-		ConvertFile converter = new ConvertFile(inputFile) ;
-		List<String> modifiedLines = converter.convertFileContents(lines, action) ;
-
-		// Write out the new file
-		outputFile.getParentFile().mkdirs() ;
-		try (PrintWriter output = new PrintWriter(new FileWriter(outputFile))) {
-			for (String outLine : modifiedLines) {
-				output.println(outLine) ;
-			}
-		};
-	}
-
+    public enum Action { kCopy, kConvert}
+
+    public static void main(String[] args) throws IOException {
+        System.out.println("flaxe v1.0");
+
+        if (args.length < 3) {
```

Comment:
` || args.length > 3` ??  Do you think that noticing too many args would be helpful?